requirements/dev.txt

```diff
@@ -1,18 +1,18 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
```

Comment:
If this file is autogenerated, why is it stored in the repo? Is there a way to generate it when it is needed, e.g. as an additional build step in the CI and conda jobs? Same question for `main.txt`.

Reply:
A given version of the input files can easily lead to broken software as dependencies evolve. One of the harder parts of making any update to this package is dealing with this breakage.

Thus I consider it crucial to commit the generated files, because this pins the exact versions of software that the service will use, leaving the code in a known-to-be-working state.